src/core/providers/strategies/helpers/signMessage/signMessage.ts

```diff
@@ -0,0 +1,51 @@
+import { Message } from '@multiversx/sdk-core/out';
+import { PendingTransactionsEventsEnum } from 'core/managers/internal/PendingTransactionsStateManager/types/pendingTransactions.types';
+import { SigningWarningsEnum } from 'types/enums.types';
+import { getModalHandlers } from '../getModalHandlers';
+
+type SignMessageWithModalPropsType<T> = {
```

Comment:
Maybe this could be written as an interface? Just a suggestion, it's not wrong like this either.